blocks/blockstore/blockstore.go

```diff
@@ -112,11 +132,8 @@ func (bs *blockstore) Get(k key.Key) (blocks.Block, error) {
 func (bs *blockstore) Put(block blocks.Block) error {
 	k := block.Key().DsKey()
 
-	// Has is cheaper than Put, so see if we already have it
-	exists, err := bs.datastore.Has(k)
-	if err == nil && exists {
-		return nil // already stored.
-	}
+	// Note: The Has Check is now done by the MultiBlockstore
```

Comment:
Are you sure about that? MultiBlockstore just calls Has method of Blockstore but it doesn't mean that Blockstore (on Datastore) will check the Has of datastore.